engine/src/main/java/org/terasology/recording/RecordedEventSerializer.java

```diff
@@ -17,509 +17,103 @@
-                TypeHandler<ButtonState> buttonStateTypeHandler =
-                        typeSerializationLibrary.getTypeHandler(ButtonState.class, getClass().getClassLoader()).get();
-                ButtonState state = buttonStateTypeHandler.deserializeOrThrow(data);
-                JsonObject aux = jsonObject.get("mousePosition").getAsJsonObject();
-                Vector2i mousePosition = new Vector2i(aux.get("x").getAsInt(), aux.get("y").getAsInt());
-                float delta = jsonObject.get("delta").getAsFloat();
-                MouseButtonEvent event = new MouseButtonEvent(button, state, delta);
-                event.setMousePosition(mousePosition);
-                newEvent = event;
-            } else if (clazz.equals(MouseAxisEvent.class)) {
-                GsonPersistedData data = new GsonPersistedData(jsonObject.get("mouseAxis"));
-                TypeHandler<MouseAxisEvent.MouseAxis> typeHandler =
-                        typeSerializationLibrary.getTypeHandler(MouseAxisEvent.MouseAxis.class, getClass().getClassLoader()).get();
-                MouseAxisEvent.MouseAxis mouseAxis = typeHandler.deserializeOrThrow(data);
-                float value = jsonObject.get("value").getAsFloat();
-                float delta = jsonObject.get("delta").getAsFloat();
-                MouseAxisEvent aux = MouseAxisEvent.create(mouseAxis, value, delta);
-                newEvent = MouseAxisEvent.createCopy(aux);
-            } else if (clazz.equals(MouseWheelEvent.class)) {
-                JsonObject aux = jsonObject.get("mousePosition").getAsJsonObject();
-                Vector2i mousePosition = new Vector2i(aux.get("x").getAsInt(), aux.get("y").getAsInt());
-                int wheelTurns = jsonObject.get("wheelTurns").getAsInt();
-                float delta = jsonObject.get("delta").getAsFloat();
-                newEvent = new MouseWheelEvent(mousePosition, wheelTurns, delta);
-            } else {
-                logger.error("Not an Input Event");
-                return null;
-            }
-
-            if (newEvent instanceof BindButtonEvent) {
-                bindButtonEventSetup((BindButtonEvent) newEvent, jsonObject);
-            } else if (newEvent instanceof BindAxisEvent) {
-                bindAxisEvent((BindAxisEvent) newEvent, jsonObject);
-            }
+        try (Reader reader = new FileReader(filePath)) {
+            Gson gson = new GsonBuilder().create();
+            JsonElement jsonElement = gson.fromJson(reader, JsonElement.class);
+            PersistedData persistedData = new GsonPersistedData(jsonElement);
 
-            inputEventSetup(newEvent, jsonObject);
-        } catch (Exception e) {
-            logger.error("Error while deserializing event. Could not find class " + className, e);
+            Optional<List<RecordedEvent>> recordedEvents = recordedEventListTypeHandler.deserialize(persistedData);
+            recordedEvents.ifPresent(events::addAll);
```

Comment:
Can we log an error if `recordedEvents` is `Optional.empty()` (`!recordedEvents.isPresent()`) here? If `Optional.empty()` is returned by `TypeHandler.deserialize`, it means there was some problem during deserialization. If there are no recorded events, `deserialize` will return an `Optional` containing an empty list.

Reply:
I just added the log.